src/affine_dimensions.jl

```diff
@@ -0,0 +1,452 @@
+
+const AbstractQuantityOrArray{T,D} = Union{UnionAbstractQuantity{T,D}, QuantityArray{T,<:Any,D}}
+
+abstract type AbstractAffineDimensions{R} <: AbstractDimensions{R} end
+
+const AffineOrSymbolicDimensions{R} = Union{AbstractAffineDimensions{R}, AbstractSymbolicDimensions{R}}
+
+"""
+    AffineDimensions{R}(scale::Float64, offset:Float64, basedim::Dimensions{R}, symbol::Symbol=nothing)
+
+AffineDimensions adds a scale and offset to Dimensions{R} allowing the expression of affine transformations of units (for example °C)
+The offset parameter is in SI units (i.e. having the dimension of basedim)
+"""
+@kwdef struct AffineDimensions{R} <: AbstractAffineDimensions{R}
+    scale::Float64 = 1.0
+    offset::Float64 = 0.0
+    basedim::Dimensions{R}
+    symbol::Symbol = :nothing
+end
```

Comment:
Could you please avoid using `@kwdef` here? See how the other structs are manually defined. I think `@kwdef` is a bit dangerous because it permits too many possible constructors, many of which we don't actually want